vault/external_tests/token/batch_token_test.go

```diff
@@ -331,90 +331,192 @@ func TestTokenStore_Roles_Batch(t *testing.T) {
 	var err error
 	var secret *api.Secret
 
-	_, err = client.Logical().Write("auth/token/roles/testrole", map[string]interface{}{
-		"bound_cidrs": []string{},
-	})
-	if err != nil {
-		t.Fatal(err)
-	}
-	secret, err = client.Auth().Token().CreateWithRole(&api.TokenCreateRequest{
-		Policies: []string{"default"},
-		Type:     "batch",
-	}, "testrole")
-	if err != nil {
-		t.Fatal(err)
-	}
-	client.SetToken(secret.Auth.ClientToken)
-	_, err = client.Auth().Token().LookupSelf()
-	if err != nil {
-		t.Fatal(err)
-	}
-	if secret.Auth.ClientToken[0:2] == "b." {
-		t.Fatal(secret.Auth.ClientToken)
+	// Test service
+	{
+		_, err = client.Logical().Write("auth/token/roles/testrole", map[string]interface{}{
+			"bound_cidrs": []string{},
+			"token_type":  "service",
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		secret, err = client.Auth().Token().CreateWithRole(&api.TokenCreateRequest{
+			Policies: []string{"default"},
+			Type:     "batch",
+		}, "testrole")
+		if err != nil {
```

Comment:
Shouldn't this fail as the client is requesting a "batch" token while the role is set to "service"?

Reply:
Oh, the client setting will be ignored, if the role doesn't allow it. Nvm.